Whiskey/Functions/New-WhiskeyContext.ps1

```diff
@@ -8,19 +8,28 @@ function New-WhiskeyContext
     .DESCRIPTION
     The `New-WhiskeyContext` function creates a context object used when running builds. It gets passed to each build task. The YAML file at `ConfigurationPath` is parsed. If it has a `Version` property, it is converted to a semantic version, a classic version, and a NuGet verson (a semantic version without any build metadata). An object is then returned with the following properties:
 
-    * `ConfigurationPath`: the absolute path to the YAML file passed via the `ConfigurationPath` parameter
+    * `ApiKeys`: any api keys needed that need to be referenced by tasks are stored here with a unique ID. Use `Add-WhiskeyApiKey` to add api keys to this property.
     * `BuildRoot`: the absolute path to the directory the YAML configuration file is in.
-    * `OutputDirectory`: the path to a directory where build output, reports, etc. should be saved. This directory is created for you.
-    * `Version`: a `SemVersion.SemanticVersion` object representing the semantic version to use when building the application. This object has two extended properties: `Version`, a `Version` object that represents the semantic version with all pre-release and build metadata stripped off; and `ReleaseVersion` a `SemVersion.SemanticVersion` object with all build metadata stripped off.
-    * `ReleaseVersion`: the semantic version with all build metadata stripped away, i.e. the version and pre-release only.
-    * `Configuration`: the parsed YAML as a hashtable.
-    * `DownloadRoot`: the path to a directory where tools can be downloaded when needed. 
     * `ByBuildServer`: a flag indicating if the build is being run by a build server.
     * `ByDeveloper`: a flag indicating if the build is being run by a developer.
-    * `ApplicatoinName`: the name of the application being built.
+    * `BuildMetadata`: metadata information of the current build on a build server.

```

Comment:
Some of these properties are private and shouldn't be used outside of Whiskey. Remove:

* Configuration
* Credentials
* DownloadRoot
* Task Defaults
* Variables